src/picongpu/include/plugins/particleMerging/ParticleMerger.kernel

```diff
@@ -0,0 +1,422 @@
+                            voronoiCell.getMaxMomentumSpread2( splittingComponent ) > momSpreadThreshold2
+                        )
+                    )
+                    {
+                        /* create two new sub Voronoi cells */
+                        voronoiCell.setToSplitting(
+                            splittingComponent,
+                            voronoiIndexPool.get(), /* lower Voronoi cell id */
+                            voronoiIndexPool.get()  /* higher Voronoi cell id */
+                        );
+
+                        /* abort if Voronoi index pool is full */
+                        if( voronoiCell.lowerCellId == -1 || voronoiCell.higherCellId == -1 )
+                        {
+                            voronoiCell.setToAbort();
+
+                            break;
+                        }
+
+                        /* initialize the two new sub Voronoi cells in `collecting` state */
+                        listVoronoiCells[voronoiCell.lowerCellId] = VoronoiCell( voronoiCell.splittingStage );
+                        listVoronoiCells[voronoiCell.higherCellId] = VoronoiCell( voronoiCell.splittingStage );
+
+                        break;
+                    }
+
+                    /* switch to momentum-splitting-stage after position-splitting-stage */
+                    if( voronoiCell.splittingStage == VoronoiSplittingStage::position )
+                    {
+                        voronoiCell = VoronoiCell( VoronoiSplittingStage::momentum );
+
+                        break;
+                    }
+
+                    /* if the Voronoi cell is neither too small in count
+                     * nor too large in spread of position or momentum
+                     * nor too low in mean energy it is ready to be merged
+                     */
+                    voronoiCell.setToReadyForMerging();
+
+                    break;
+                }
+
+            default:
+                /* delete Voronoi cell */
+                voronoiIndexPool.release( voronoiCellId );
+
+                break;
+            }
+        }
+    }
+
+
+    /** Entry point of the particle merging algorithm
+     *
+     * @param cellIndex n-dim. cell index from the origin of the local domain
+     */
+    DINLINE void operator()( const PMacc::math::Int<simDim>& cellIndex )
+    {
+        /* multi-dim vector from origin of the super cell to a cell in units of cells */
+        const PMacc::math::Int<simDim> threadIndex = cellIndex % SuperCellSize::toRT();
+
+        /* conversion from a multi-dim cell coordinate to a linear coordinate
+         * of the cell in its super cell */
+        const int linearThreadIdx = PMacc::math::linearize(
+            PMacc::math::CT::shrinkTo<SuperCellSize, simDim-1>::type::toRT(),
+            threadIndex
+        );
+
+        /* number of initial Voronoi cells */
+        PMACC_SMEM( numInitialVoronoiCells, uint16_t );
```

Comment:
Pow can also be calculated at compile time, we hat already a compile time pow implemented in [radioation utilities](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/plugins/radiation/utilities.hpp#L86) but this function is not exposed to PMacc (#1652).
For this special case you can use `1<<simDim` and add a comment that this is equal to `pow(2,simDim)` than it removes a shared memory variable which speedups the code if we go to cpus.